chia/full_node/coin_store.py

```diff
@@ -190,7 +189,7 @@ async def get_coin_record(self, coin_name: bytes32) -> CoinRecord | None:
                 if row is not None:
                     coin = self.row_to_coin(row)
                     spent_index = uint32(0) if row[1] <= 0 else uint32(row[1])
-                    return CoinRecord(coin, row[0], spent_index, row[2], row[6])
+                    return CoinRecord(coin, row[0], spent_index, row[2] != 0, row[6])
```

Comment:
is `row[2]` indicating that the coin is spent? If so, I think this needs to be `> 0` now. Remember we re-purposed negative numbers